src/lib/viewers/image/ImageBaseViewer.js

```diff
@@ -86,6 +94,36 @@ class ImageBaseViewer extends BaseViewer {
         });
     }
 
+    /**
+     * Gets the viewport dimensions.
+     *
+     * @return {Object} the width & height of the viewport
+     */
+
+    getViewportDimensions() {
+        return {
+            width: this.wrapperEl.clientWidth - 2 * IMAGE_PADDING,
+            height: this.wrapperEl.clientHeight - 2 * IMAGE_PADDING,
+        };
+    }
+
+    /**
+     * Sets mode to be AnnotationMode.NONE if the zoomed image overflows the viewport.
+     *
+     * @return {void}
+     */
+    handleZoomEvent({ newScale }) {
+        const width = newScale[0];
+        const height = newScale[1];
+
+        const viewport = this.getViewportDimensions();
+
+        // If the image is overflowing the viewport, we should set annotation mode to be NONE so that the user can pan
+        if (width > viewport.width || height > viewport.height) {
+            this.processAnnotationModeChange(this.annotationControlsFSM.transition(AnnotationInput.RESET));
+        }
+    }
```

Comment:
These seem to be `ImageViewer` specific, i.e. `IMAGE_PADDING` may not apply to the `MultiImageViewer` and similarly, the `handleZoomEvent` should only apply to the `ImageViewer`